tests/unit_tests/dft/include/test_common.hpp

```diff
@@ -0,0 +1,228 @@
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_TEST_COMMON_HPP
+#define ONEMKL_TEST_COMMON_HPP
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+template <typename T>
+struct complex_info {
+    using real_type = T;
+    static const bool is_complex = false;
+};
+
+template <typename T>
+struct complex_info<std::complex<T>> {
+    using real_type = T;
+    static const bool is_complex = true;
+};
+
+template <typename T>
+constexpr bool is_complex() {
+    return complex_info<T>::is_complex;
+}
+template <typename T>
+constexpr int num_components() {
+    return is_complex<T>() ? 2 : 1;
+}
+
+template <typename fp>
+typename std::enable_if<!std::is_integral<fp>::value, bool>::type check_equal(fp x, fp x_ref,
+                                                                              int error_mag) {
+    using fp_real = typename complex_info<fp>::real_type;
+    fp_real bound = (error_mag * num_components<fp>() * std::numeric_limits<fp_real>::epsilon());
+
+    bool ok;
+
+    fp_real aerr = std::abs(x - x_ref);
+    fp_real rerr = aerr / std::abs(x_ref);
+    ok = (rerr <= bound) || (aerr <= bound);
+    if (!ok) {
+        std::cout << "relative error = " << rerr << " absolute error = " << aerr
+                  << " limit = " << bound << std::endl;
+    }
+    return ok;
+}
+
+template <typename fp>
+bool check_equal(fp x, fp x_ref, int error_mag, std::ostream &out) {
+    bool good = check_equal(x, x_ref, error_mag);
+
+    if (!good) {
+        out << "Difference in result: DPC++ " << x << " vs. Reference " << x_ref << std::endl;
+    }
+    return good;
+}
+
+template <typename fp>
+bool check_equal_vector(fp *v, fp *v_ref, int n, int inc, int error_mag, std::ostream &out) {
+    constexpr size_t max_print = 20;
+    int abs_inc = std::abs(inc), count = 0;
+    bool good = true;
+
+    for (int i = 0; i < n; i++) {
+        if (!check_equal(v[i * abs_inc], v_ref[i * abs_inc], error_mag)) {
+            int i_actual = (inc > 0) ? i : n - i;
+            std::cout << "Difference in entry " << i_actual << ": DPC++ " << v[i * abs_inc]
+                      << " vs. Reference " << v_ref[i * abs_inc] << std::endl;
+            good = false;
+            count++;
+            if (count > max_print) {
+                return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+template <typename vec1, typename vec2>
+bool check_equal_vector(vec1 &v, vec2 &v_ref, int n, int inc, int error_mag, std::ostream &out) {
```

Comment:
I think the only difference between the two versions of check_equal_vector is that this one can have a difference vec type, but that seems useless since check_equal expects both arguments to be of the same type. I think this should be deleted.

Reply:
I've removed the pointer version in [56d18e6](https://github.com/oneapi-src/oneMKL/pull/261/commits/56d18e60d45dc3a0b617eef414eb0e84cbe6d7c3).